### PR TITLE
Task: Baseline CI - Add Github workflow to build Baseline profile

### PR DESCRIPTION
**Pull request by jarroyoesp**
### Pull request checklist

- [x] I have **rebased** this branch on top of the destination branch (usually `master`)
- [x] I have executed locally `./gradlew spotlessApply check` *before creating the commit* and it has run successfully
- [x] My contribution is fully baked and ready to be merged as is
- [x] I have performed a self-review of my own code
- [x] These changes are in compliance with [the coding style](/CODING_STYLE.md) of this project
- [x] There are no `WIP` commits in this PR
- [x] If there are changes to the UI, I have uploaded at least one screenshot of every UI change

### Type of Changes

|     | Type                   |
|-----|------------------------|
|   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
|    | :hammer: Refactoring   |
|     | :scroll: Docs          |

### Screenshots
NO UI changes

### CODE REVIEW

File: .github/workflows/baseline-profile-generator.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+name: Baseline Profile generator
+
+on:
+  pull_request:
```
[the remark this answers is not in the document]
Comment (jarroyoesp) [reply .github/workflows/baseline-profile-generator.yml:4]: Yep, now it is just to test it

File: .github/ci-gradle.properties [hunk 2]
```diff
@@ -19,5 +19,12 @@ org.gradle.workers.max=2
 # disable annoying Gradle Welcome in CI/CD
 org.gradle.welcome=never
 
+android.experimental.enableExperimentalOptions=true
+android.experimental.testOptions.managedDevices.allowOldApiLevelDevices=true
+android.experimental.testOptions.managedDevices.emulator.showKernelLogging=true
+android.experimental.testOptions.managedDevices.maxConcurrentDevices=1
+android.experimental.testOptions.managedDevices.setupTimeoutMinutes=90
+android.testoptions.manageddevices.emulator.gpu=swiftshader_indirect
+
```
Comment (jarroyoesp) [inline .github/ci-gradle.properties:28]: 👍

File: .github/workflows/generate-baseline-profile.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Accept Android licenses
+        run: yes | "$ANDROID_HOME"/cmdline-tools/latest/bin/sdkmanager --licenses || true
+
+      - name: Generate Baseline Profile
+        run: ./gradlew :macrobenchmark:forlago:pixel6Api31BenchmarkAndroidTest
+          -Pandroid.testInstrumentationRunnerArguments.androidx.benchmark.enabledRules=baselineprofile
+
+      - name: Move & Rename Baseline Profile
+        run: mv -f macrobenchmark/forlago/build/outputs/managed_device_android_test_additional_output/benchmark/pixel6Api31/BaselineProfileGenerator_generate-baseline-prof.txt  apps/forlago/src/main/baselineProfiles/baseline-prof.txt
```
Comment (jarroyoesp) [inline .github/workflows/generate-baseline-profile.yml:61]: 👍 relative path

File: build-conventions/src/main/kotlin/com/leinardi/forlago/ext/ProjectExt.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.gradle.kotlin.dsl.withType
+
+fun Project.configureJavaCompile() {
+    plugins.withType<JavaPlugin> {
+        tasks.withType<JavaCompile>().configureEach {
+            options.compilerArgs.add("-Xlint:deprecation")
+            options.compilerArgs.add("-Xlint:unchecked")
+        }
+    }
+}
```
Comment (jarroyoesp) [inline build-conventions/src/main/kotlin/com/leinardi/forlago/ext/ProjectExt.kt:31]: why is this needed?

File: build-conventions/src/main/kotlin/forlago.android-conventions.gradle.kts [hunk 5]
```diff
@@ -76,6 +77,8 @@ plugins.withType<AndroidBasePlugin>().configureEach {
                 )
                 jvmTarget = config.android.javaVersion.get().toString()
             }
+
+            configureGradleManagedDevices(minApiLevel = config.android.minSdk.get(), maxApiLevel = config.android.targetSdk.get())
```
Comment (jarroyoesp) [inline build-conventions/src/main/kotlin/forlago.android-conventions.gradle.kts:81]: Nice solution to have GMDs with all the possible APIs created 👏🏻

File: build-conventions/src/main/kotlin/com/leinardi/forlago/ext/ProjectExt.kt (hunk 4 above)
Comment (jarroyoesp) [reply build-conventions/src/main/kotlin/com/leinardi/forlago/ext/ProjectExt.kt:31]: Now I've seen you moved to a extension function, but why is only needed on `library conventions`?

Comment (leinardi) [reply build-conventions/src/main/kotlin/com/leinardi/forlago/ext/ProjectExt.kt:31]: it is applied to the base android convention and to the base kotlin conventio, so it is applied to every project. The reason is this: https://discuss.gradle.org/t/basepluginconvention-deprecation-warning-for-archivefilename/46269/2

Comment (jarroyoesp) [reply build-conventions/src/main/kotlin/com/leinardi/forlago/ext/ProjectExt.kt:31]: True, I didnt see it on `android-conventions` 👍

**Review by leinardi — APPROVED** [review]
